echopop/extensions/diagnostics.py

```diff
@@ -0,0 +1,510 @@
+        lines_gdf = (
+            gdf.groupby("transect_num")
+            .apply(
+                lambda x: LineString(x.geometry.tolist()),
+                include_groups=False,
+            )
+            .reset_index(name="geometry")
+        )
+        # ---- Convert back to GeoDataFrame
+        transect_gdf = gpd.GeoDataFrame(lines_gdf)
+
+        # Aggregate the data
+        DATACOPY = DATAINPUT.loc[DATAINPUT.nasc > 0].copy()
+        # ---- Apply min-max scaling
+        DATACOPY["nasc_scale"] = (DATACOPY.nasc - DATACOPY.nasc.min()) / (
+            DATACOPY.nasc.max() - DATACOPY.nasc.min()
+        )
+        # ---- Set sizing
+        DATACOPY["nasc_size"] = DATACOPY["nasc_scale"] * 25 + 2  # (minimum size value)
+
+        # Create TOOLTIP
+        TOOLTIPS = """
+            <div style="position: relative;">
+                <style>
+                    div.bk-tooltip-content > div > div:not(:first-child) {{
+                        display:none !important;
+                    }}
+                </style>
+                <div>
+                    <b>Transect: </b> <span>@transect_num</span> <br>
+                    <b>NASC: </b> <span>@nasc</span> m^2 nmi^-2 <br>
+                </div>
+            </div>
+            """
+        # ---- Create hovertool
+        hover_tool = HoverTool(tooltips=TOOLTIPS, mode="mouse", point_policy="snap_to_data")
+
+        # Create transect layer
+        transect_gv = gv.Path(transect_gdf, label="Transect").opts(color="black", show_legend=True)
+
+        # Create point layer
+        points_gv = gv.Points(
+            DATACOPY,
+            kdims=["longitude", "latitude"],
+            vdims=["transect_num", "nasc", "nasc_scale", "nasc_size"],
+        ).opts(
+            # ---- COLOR
+            colorbar=True,
+            colorbar_position="right",
+            cmap="reds",
+            color="nasc_scale",
+            clim=(0, 1),
+            colorbar_opts={
+                "title": "Min-max scaled NASC [0, 1]",
+                "title_text_font_style": "bold",
+                "title_text_color": "black",
+                "major_label_text_color": "black",
+            },
+            # AESTHETICS
+            line_color="black",
+            line_width=0.25,
+            size="nasc_size",
+            tools=[hover_tool],
+        )
+
+        # Stack all of the layers to complete the plot
+        full_plot = (transect_gv * self.tile * points_gv).opts(
+            height=800, width=800, xlabel="Longitude (\u00B0E)", ylabel="Latitude (\u00B0N)"
+        )
+        # ---- Return plot
+        return full_plot
+
+    def stratified_results(self) -> pn.Column:
+
+        # Get the full datasets
+        # ---- Get stratification information
+        strata_info = self.data.input["spatial"]["inpfc_strata_df"].copy()
+        # ---- Initialize
+        DATAINPUT = {"Biomass": {}, "Biomass density": {}}
+        # ---- Kriging
+        if "kriging" in self.data.results:
```

Comment:
Do you anticipate cases when "kriging" or "transect" is NOT in `self.data.results`?

Reply:
Good point. For this function to run, I think it would only be reasonable to anticipate `"kriging"` to be missing since it inherits values from `"transect"`. I can add a warning message here that states as much.